replica/table.cc

```diff
@@ -2379,7 +2394,7 @@ table::as_mutation_source_excluding(std::vector<sstables::shared_sstable>& ssts)
     });
 }
 
-class table::table_state : public compaction::table_state {
+class compaction_group::table_state : public compaction::table_state {
```

Comment:
How did it compile if it's defined after it's used?

Reply:
Both GCC and clang allow me to instantiate the *ctor* of class that wasn't defined yet. Had an member been accessed instead, then the compiler would yell at me.

This is what we have:
https://godbolt.org/z/4eTs8n73P

Reply:
With make_unique<T>(args...), we satisfy the compiler, then the class is found defined by the time the template is evaluated.

Reply:
Interesting reading about this topic: https://howardhinnant.github.io/incomplete.html

Reply:
ok